### PR TITLE
Intersects_protection(): Move from Minetest Game to builtin

### PR DESCRIPTION
**Pull request by paramat**
A useful function that applies 'core.is_protected()' to a 3D lattice of
points evenly spaced throughout a defined volume, with a parameter for
the maximum spacing of points.
////////////////

Attends to #6533 
A later PR will remove the function from MTGame.
Mostly copypaste but i made 'interval' default to a reasonable value of 4 if left out, to avoid excessive checks.
@SmallJoker

### CODE REVIEW

File: builtin/game/misc.lua [hunk 1]
```diff
@@ -40,14 +40,17 @@ function core.check_player_privs(name, ...)
 	return true, ""
 end
 
+
```
Comment (red-001) [inline builtin/game/misc.lua]: Also add an empty line here?

**Comment by raymoo** [comment]
Is this intended to be overridden by mods who can provide a more efficient and exact protection intersection-checking mechanism? Such as using an area store?

**Comment by SmallJoker** [comment]
@raymoo That was already mentioned almost a week ago: https://github.com/minetest/minetest/commit/01bc817fe0a59e2e2b20c26d395c5b989c5156c9#commitcomment-27049889

**Comment by paramat** [comment]
https://github.com/minetest/minetest_game/pull/2030 merged.

**Pull request merged**